xcode/Subconscious/Shared/Components/Common/Forms/ValidatedTextField.swift

```diff
@@ -11,34 +11,38 @@ import SwiftUI
 struct ValidatedTextField: View {
     var placeholder: String
     @Binding var text: String
+    var onFocusChanged: ((Bool) -> Void)?
     var caption: String
-    var isValid: Bool = true
-
+    var hasError: Bool = false
+    
```

Comment:
Validation and error presentation are seperate concerns here, since we should suppress errors on fields the user hasn't yet edited BUT those fields are technically invalid. Once they've been touched (or the user tries to submit the form without entering any values) _then_ we should actually display the error state.